src/GraphQL/Validation/ValidationContext.cs

```diff
@@ -318,21 +318,14 @@ object ParseValue(IGraphType type, VariableDefinition variableDef, VariableName
             // Coerces a scalar value
             object ParseValueScalar(ScalarGraphType scalarGraphType, VariableDefinition variableDef, VariableName variableName, object value)
             {
-                if (value == null)
-                    return null;
-
-                object ret;
-
                 try
                 {
-                    ret = scalarGraphType.ParseValue(value);
+                    return scalarGraphType.ParseValue(value);
```

Comment:
So, it's absolutely OK now:
1. Return `null` as parsed scalar result.
2. Accept `null` and return _some_ parsed result not necessarily equal to `null`.

If so, this must be described in detail in the documentation.

Reply:
Yes, we will need a significant change to the documentation for scalars.  We could perhaps provide two samples: one where empty strings internally are coerced to null externally, and one where 0 values internally are converted to/from null external values.